ministryofjustice/correspondence_tool_staff
# CT-3225 change complaint case type

## Description
makes the case status reset when the complaint_case type changes, also shows multiple contact details for different complaint types if they're present.

**Notes for reviewer:**
- Do you now if there is a cleaner / more API way to reset the case status? I just amend it on the loaded ActiveRecord before its re-saved but I wonder if there are better ways to do this? 🤔 

## Self-review checklist


### Screenshots


### Related JIRA tickets



### Manual testing instructions

Try to change the complaint case_type to prove the following
- changing between case types should reset the status to `to_be_assessed`
- changing the contact details of the complaint type but not changing the actual case type should not reset the status
- changing sub types or priorities should not change the status.
- changing the status twice (i.e. standard --> ico --> litigation) should mean that all of the contact details for ICO and litigation are visible on the summary screen.
- contact details should only appear if they are `.present?` no blank fields on the summary page 👍
